bftengine/include/bcstatetransfer/SimpleBCStateTransfer.hpp

```diff
@@ -76,6 +77,29 @@ class IAppState {
   // outBlock and outBlockSize. Returns true IFF block blockId exists.
   virtual bool getBlock(uint64_t blockId, char *outBlock, uint32_t *outBlockSize) = 0;
 
+  struct GetBlockContext {
```

Comment:
General question:
What happens if there are more than 2 destinations?

Reply:
Do you mean more than 1?
And in general, it is 2 up to f.
This is an advanced topic, we do not yet address and we don't have any good testing tools (manual only) to validate.
In that case, there is no optimization. I'll open a future issue for that - in general, a source should reject a destination if it already has other destination.

Reply:
This structure is redundant - an std::vector could be used for bounded data.
exists can be replaced by a std::promise::set_exception()

Reply:
Thank you for the detailed answer!